sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/auth/MySQLAuthenticationEngine.java

```diff
@@ -54,11 +54,11 @@ public void handshake(final ChannelHandlerContext context, final BackendConnecti
     @Override
     public boolean auth(final ChannelHandlerContext context, final PacketPayload payload, final BackendConnection backendConnection) {
         MySQLHandshakeResponse41Packet response41 = new MySQLHandshakeResponse41Packet((MySQLPacketPayload) payload);
-        if (authenticationHandler.login(response41.getUsername(), response41.getAuthResponse())) {
-            if (!Strings.isNullOrEmpty(response41.getDatabase()) && !LogicSchemas.getInstance().schemaExists(response41.getDatabase())) {
-                context.writeAndFlush(new MySQLErrPacket(response41.getSequenceId() + 1, MySQLServerErrorCode.ER_BAD_DB_ERROR, response41.getDatabase()));
-                return true;
-            }
+        if (!Strings.isNullOrEmpty(response41.getDatabase()) && !LogicSchemas.getInstance().schemaExists(response41.getDatabase())) {
+            context.writeAndFlush(new MySQLErrPacket(response41.getSequenceId() + 1, MySQLServerErrorCode.ER_BAD_DB_ERROR, response41.getDatabase()));
+            return true;
+        }
+        if (authenticationHandler.login(response41.getUsername(), response41.getAuthResponse(), response41.getDatabase())) {
```

Comment:
Will it be better if we pass `response41` to `authenticationHandler.login()` instead of `username, pwd, db`?

Reply:
Good idea.